be/src/vec/olap/vcollect_iterator.cpp

```diff
@@ -470,7 +479,32 @@ Status VCollectIterator::Level1Iterator::_merge_next(Block* block) {
             LOG(WARNING) << "next failed: " << res;
             return res;
         }
-    } while (target_block_row < _batch_size);
+        if (target_block_row >= _batch_size) {
```

Comment:
Maybe we should limit a batch not over batch_size

Reply:
target_block_row must be less than batch_size